web/src/components/users/FirstUser.jsx

```diff
@@ -166,6 +172,26 @@ export default function FirstUser() {
     setFormValues({ ...formValues, [name]: value });
   };
 
+  const suggestUsernames = (fullName) => {
```

Comment:
I'll start immediately with some uncertainties :smile:

When the user types two words for the full name, these current suggestions work well.

However, if the user's full name has more than two words (names), the choice is not so clear.
People having longer full names is a common thing in some parts of the world (like Spain) and with the current approach to suggesting the username, Ronaldo Luís Nazário de Lima would be presented with the following choices:

![image](https://github.com/openSUSE/agama/assets/112288843/da833ac8-3cdd-4de2-a419-d6ada8fdd1fb)


The issues are apparent, the first suggestion is pretty long, the second might be as well and only the third one might be okay.

Should we maybe take only the first two words (names)?
Or maybe some other approach entirely, feel free to make a suggestion.

Reply:
I see your point, but I fear that guessing which are the _right or expected parts_ is a bit difficult due to the fact on how different can be [personal names around the world](https://www.w3.org/International/questions/qa-personal-names) (long and old article).

Therefore, I wouldn't worry too much about them. They are just suggestions. Of course, the idea is to show up the more sensible ones, but we can't do so for every single user. I propose to below as first. Some of them are based on feedback received in the issue, some are not:


* Just the first word (the simplest approach)

  To make @kobliha [happy](https://github.com/openSUSE/agama/issues/112#issuecomment-1351026852)

  * Lubos Kocman: **lubos**
  * Imobach González Sosa: **imobach**
  * Ronaldo Luís Nazário de Lima: **ronaldo**

* First letter of first word plus second word

  It satisfies the [original request](https://github.com/openSUSE/agama/issues/112#issue-1189780790) from @lkocman 

  * Lubos Kocman: **lkocman**
  * Imobach González Sosa: **igonzalez**
  * Ronaldo Luís Nazário de Lima: **rluis**

* First letter of first word plus all other words

  It would suggest a username like the one @imobachgs [have at SUSE](https://github.com/openSUSE/agama/issues/112#issuecomment-1327196554)

  * Lubos Kocman: **lkocman**
  * Imobach González Sosa: **igonzalezsosa**
  * Ronaldo Luís Nazário de Lima: **rluisnazariodelima**

* First word plus first letter of the other words (maybe with a limit of just 2 more words?)

  * Lubos Kocman: **lubosk**
  * Imobach González Sosa: **imobachgs**
  * Ronaldo Luís Nazário de Lima: **ronaldolndl** (or **ronaldoln** if using a limit)

* First letter of all words except the last one, which would be complete

  * Lubos Kocman: **lkocman**
  * Imobach González Sosa: **igsosa**
  * Ronaldo Luís Nazário de Lima: **rlndlima**

* All words

  * Lubos Kocman: **luboskocman**
  * Imobach González Sosa: **imobachgonzalezsosa**
  * Ronaldo Luís Nazário de Lima: **ronaldoluiznaariodelima**

So, after removing duplicates, suggestions for these users will be, 

<table>
<thead>
<tr>
<th>Lubos Kocman</th>
<th>Imobach González Sosa</th>
<th>Ronaldo Luís Nazário de Lima</th>
</tr>
</thead>
<tbody>
<tr>
<td>

* lubos
* lkocman
* lubosk

</td>
<td>

* imobach
* igonzalez
* igonzalezsosa
* imobachgs
* igsosa
* imobachgonzalezsosa

</td>
<td>

* ronaldo
* rluis
* rluisnazariodelima
* ronaldolndl (or ronaldoln)
* rlndlima
* ronaldoluiznaariodelima

</td>
</tr>
</tdoby>
</table>

I do not have strong preference and ideas from others are welcome. I think that we can release the feature with these few (or even only the three or four first ones)  and let's see if we receive more suggestion requests.

BTW, as you can see in my examples above, I prefer not using the "special characters" in the suggested username (accents and friends). See https://stackoverflow.com/a/70288180 and https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/normalize